### ISSUE TITLE
NEXT button remains visible in the last slide

### ISSUE THREAD

**Issue by CeJienAJPC** [HeinrichReimer/material-intro#199]
I am using the latest dependency **Version 1.6.2** (with 4 Simple Slides and a custom slide)

For NEXT button I am using `setButtonNextFunction(BUTTON_NEXT_FUNCTION_NEXT);`

However, the NEXT button remains visible in the last slide (a custom slide in my case). This wasn't the case with **Version 1.6.1**. Illustrated in the below image.

![next button](https://cloud.githubusercontent.com/assets/15603039/24321740/17e7d52a-117a-11e7-90ff-0ace1aca808b.png)

**Comment by sunil19822701** [comment]
Change it to this and try
setButtonNextFunction(BUTTON_NEXT_FUNCTION_NEXT_FINISH);

**Comment by HeinrichReimer** [comment]
I'll investigate this...

**Comment by HeinrichReimer** [comment]
@CeJienAJPC Which library versison are you using?
Also it would be great if you could attach a screen record of that bug as well as the code you're using to create the slides.
If I see this right, you are not using the library's CTA button but your own implementation instead, is that correct?

**Comment by CeJienAJPC** [comment]
@heinrichreimer Sorry for the delayed response. I was offline for the week. To answer your questions, I am using Version 1.6.2 (where the above problem appears). However, Version 1.6.1 is performing as without the above issues.

I am using a custom Layout for the final slide

**Comment by CeJienAJPC** [comment]
@sunil19822701 
`setButtonNextFunction(BUTTON_NEXT_FUNCTION_NEXT_FINISH);` works fine with V 1.6.2
using `setButtonNextFunction(BUTTON_NEXT_FUNCTION_NEXT);` with V 1.6.2 produces the above issue.

What if, I want to use `setButtonNextFunction(BUTTON_NEXT_FUNCTION_NEXT);`?

**Comment by heinrichreimer** [comment]
Can you confirm this issue also appears when using the default layout?
If not, the issue may be in your layout file. Could you paste tyour layout code here?
I know that's a lot of questions but I'm trying to isolate the issue case.

**Comment by odin-** [comment]
I got 1.6.2. 

My problem was that I didn't understand that SKIP is default even though the icon clearly indicates so), 

I just added

setButtonBackFunction(BUTTON_BACK_FUNCTION_BACK);

And now it works just fine.

Thanks. 

You can close this one.

**Comment by daniel-stoneuk** [comment]
@odin- You mean #202, right?

**Comment by heinrichreimer** [comment]
@daniel-stoneuk Yep, #202 is closed, this issue will stay open until we find the issue.

**Comment by odin-** [comment]
@daniel-stoneuk @heinrichreimer sorry yes. You can delete my comments here if you want to.

**Comment by heinrichreimer** [comment]
I'll just tag them _**Off-topic:**_

**Comment by HSY229** [comment]
I have same issue.

```Java
public class IntroWalletActivity extends IntroActivity implements OnPageChangeListener {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    ...

    setButtonNextFunction(BUTTON_NEXT_FUNCTION_NEXT);
    setButtonNextVisible(true);

    ...
  }
}
```

## version
1.6.2

## device
ASUS Z01BDA

## Android version
7.1.1

## workaround
```Java
public class IntroWalletActivity extends IntroActivity implements OnPageChangeListener {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    ...

    setButtonNextFunction(BUTTON_NEXT_FUNCTION_NEXT_FINISH);
    setButtonNextVisible(true);

    ...
  }

  @Override
  public void onPageSelected(int position) {
    if (isLatestPage(position)) {
      setButtonNextVisible(false);
    } else {
      setButtonNextVisible(true);
    }
  }
}
```

**Comment by pinpong** [comment]
```
        addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
                // ignore
            }

            @Override
            public void onPageSelected(int position) {
                if (position == getCount() - 1) {
                    setButtonNextVisible(false);
                } else {
                    setButtonNextVisible(true);
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {
                // ignore
            }
        });
```
